Docs/introducing-blazor-forms.md

```diff
@@ -0,0 +1,215 @@
+
+    <button class="btn btn-primary mt-3" type="submit">Submit</button> 
+</EditForm> 
+
+@code 
+{ 
+    public Contact ContactModel { get; set; } = new(); 
+
+    void FormSubmitted()
+    { 
+        Logger.LogInformation($"Name: {ContactModel.Name}, Email: {ContactModel.Email}");
+
+        // You could call your Web API endpoints or services here
+        // Eg: await Http.SendJsonAsync(HttpMethod.Post, "/api/Contact/Create", ContactModel);
+    }
+}
+```
+
+The structure of the mark-up looks very similar for both the Blazor and MVC code. The key points to highlight in the Blazor code are: 
+
+* **EditForm** defines a Blazor form which renders a `<form>` element under-the-hood. You could bind either a Model (a POCO) or an `EditContext` as the data that represents the form. More on this will be discussed later. 
+
+* **Model** attribute lets us bind a top-level model object to the form. This model object is updated whenever we make changes to the form. 
+
+* **OnSubmit** is a callback event that gets triggered when we submit the form. Blazor also provides us `OnValidSubmit` and `OnInvalidSubmit` convenience methods. 
+
+* **InputText** is a Blazor form component which represents an `<input>` element. The Input element is annotated with the `@bind-Value` attribute to bind to the model. In our case it is `Model.Name` and `Model.Email`. By default the corresponding model field that's bound to this component is updated in the `onChangeEvent` handler. 
+
+The Blazor @code block initializes our ContactModel object and has a method that gets triggered when the form is submitted. Note that all code is written using C# and we can even use .NET types such as HttpClient and take advantage of .NET dependency injection to help with separation of concerns and managing dependencies.
+
+## Blazor Component Model
+
+We can make use of standard HTML form and input elements to create a Blazor form. However, it would be cumbersome to maintain their events and manage the form state when it comes to validation. Therefore, Blazor provides us a suite of out-of-the-box components which makes form validation easier.
+
+In the example above, we used the EditForm and InputText components to capture input and process the form. All Blazor form elements we have seen up to now, including the [EditForm](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web/src/Forms/EditForm.cs), inherit from an abstract class called [ComponentBase](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.componentbase?view=aspnetcore-5.0).
+
+This means any form components participate in the component lifecycle can be used to extend functionality by providing our own implementation for things such as validation and component styling if need be. We will look at a whole variety of custom controls others have built leveraging this concept.
+
+The default set of Blazor form components that map to standard HTML input controls as shown in the following table:
```

Comment:
I think we can simplify the first two paragraphs here and maybe even consider if we need the entire table of Input components to be displayed.